### PR TITLE
MR-649 Contact name for online repair

### PR DESCRIPTION
**Pull request by Asura5**
- Add component to contact details page
- Modify summary to reflect new data
- Modify 'Change' parameter to include new data

Ticket: https://hackney.atlassian.net/browse/MR-649

image

### CODE REVIEW

File: compoments/report-repair/contact-person.js [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
+        title="Who should we contact, if we need to get in touch?"
         buttonText={'Continue'}
         long={true}
         onKeyPress={phoneOnKeyPress}
-      ></TextInput>
+        nameName={'contact-name'}
+        nameType="text"
+        nameLabel="Contact  name"
+        nameValue={values.contactPerson?.contactNameValue}
+      ></ContactDetailsInput>
```
Comment (LBHCallumM) [inline compoments/report-repair/contact-person.js:28]: Could some of these attributes be moved into the component? I suspect that this component won't ever be reused.

**Review by LBHCallumM — APPROVED** [review]
Approved with comments

File: compoments/contactDetailsInput.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
+import PropTypes from 'prop-types';
+import Button from './button';
+import React from 'react';
+import {Component} from 'react';
+
+class ContactDetailsInput extends Component {
+  constructor(props) {
+    super(props);
```
[the remark this answers is not in the document]
Comment (Asura5) [reply compoments/contactDetailsInput.js:8]: To keep it consistent with the other Textbox-based inputs in this repo, namely [TextInput](https://github.dev/LBHackney-IT/housing-repairs-online-frontend/blob/6d9d49de3da90f41e84dc190281dd4f1d4ce7b0b/compoments/textInput.js#L1)

File: compoments/report-repair/contact-person.js (hunk 1 above)
Comment (Asura5) [reply compoments/report-repair/contact-person.js:28]: Good point - will move

**Pull request merged**